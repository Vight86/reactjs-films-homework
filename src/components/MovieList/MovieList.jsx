import React from 'react';
import propTypes from 'prop-types';
import MovieCard from '../MovieCard/index';
import styles from './MovieList.scss';

const MovieList = ({
  movies, genres, isGrid, movieTrailerKey, updateMovieTrailerKey,
}) => {
  const list = movies.map((movie) => (
    <MovieCard
      key={movie.id}
      movie={movie}
      genres={genres}
      isGrid={isGrid}
      movieTrailerKey={movieTrailerKey}
      updateMovieTrailerKey={updateMovieTrailerKey}
    />
  ));

  return (
    <section className={`${isGrid ? styles.movieListGrid : styles.movieListTable}`}>
      {list}
    </section>
  );
};

MovieList.propTypes = {
  movies: propTypes.arrayOf(propTypes.object).isRequired,
  genres: propTypes.arrayOf(propTypes.object).isRequired,
  isGrid: propTypes.bool.isRequired,
  movieTrailerKey: propTypes.string,
  updateMovieTrailerKey: propTypes.func.isRequired,
};

MovieList.defaultProps = {
  movieTrailerKey: null,
};

export default MovieList;
