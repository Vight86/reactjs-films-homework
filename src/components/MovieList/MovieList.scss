@import '../../utils/variables';

.movieListGrid {
  display: grid;
  margin-bottom: 4em;
  gap: 4em;
  grid-template-columns: repeat(4, 1fr);
  grid-template-rows: auto;

  & > * {
    justify-items: stretch;
  }

  @media screen and (max-width: $bp-medium) {
    grid-template-columns: repeat(2, 1fr);
    grid-template-rows: auto;
  }

  @media screen and (max-width: $bp-xxxsmall) {
    grid-template-columns: 1fr;
    grid-template-rows: auto;
  }

  @media screen and (max-width: $bp-xxxxsmall) {
    grid-template-columns: 1fr;
    grid-template-rows: auto;
  }
}

.movieListTable {
  display: block;
  margin-bottom: 4em;
}
