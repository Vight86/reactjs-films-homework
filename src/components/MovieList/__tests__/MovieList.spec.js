import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import MovieList from '../index';

const mockMovie = {
  adult: false,
  backdropPath: '/test.jpg',
  genreIds: [1, 2],
  id: 1,
  originalLanguage: 'en',
  originalTitle: 'test',
  overview: 'test',
  popularity: 1,
  posterPath: '/test.jpg',
  releaseDate: '1',
  title: 'test',
  video: false,
  voteAverage: 1,
  voteCount: 1,
};

const mockMovies = Array(2).fill(mockMovie);

const mockGenres = [{
  id: 1,
  name: 'Action',
},
{
  id: 2,
  name: 'SciFi',
}];

const mockMovieTrailerKey = 'test';
const mockUpdateMovieTrailerKey = jest.fn(() => 'tested');

let renderer;
let result;
beforeEach(() => {
  renderer = new ShallowRenderer();
  renderer.render(
    <MovieList
      movies={mockMovies}
      genres={mockGenres}
      isGrid={false}
      movieTrailerKey={mockMovieTrailerKey}
      updateMovieTrailerKey={mockUpdateMovieTrailerKey}
    />,
  );
  result = renderer.getRenderOutput();
});

afterEach(() => {
  renderer = null;
  result = null;
});

describe('render MovieList component', () => {
  it('render table layout correctly', () => {
    expect(result).toMatchSnapshot();
  });

  it('render grid layout correctly', () => {
    renderer.render(
      <MovieList
        movies={mockMovies}
        genres={mockGenres}
        isGrid
        movieTrailerKey={mockMovieTrailerKey}
        updateMovieTrailerKey={mockUpdateMovieTrailerKey}
      />,
    );
    result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
