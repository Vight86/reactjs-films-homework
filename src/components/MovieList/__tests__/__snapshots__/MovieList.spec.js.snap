// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render MovieList component render grid layout correctly 1`] = `
<section
  className="movieListGrid"
>
  <MovieCard
    genres={
      Array [
        Object {
          "id": 1,
          "name": "Action",
        },
        Object {
          "id": 2,
          "name": "SciFi",
        },
      ]
    }
    isGrid={true}
    movie={
      Object {
        "adult": false,
        "backdropPath": "/test.jpg",
        "genreIds": Array [
          1,
          2,
        ],
        "id": 1,
        "originalLanguage": "en",
        "originalTitle": "test",
        "overview": "test",
        "popularity": 1,
        "posterPath": "/test.jpg",
        "releaseDate": "1",
        "title": "test",
        "video": false,
        "voteAverage": 1,
        "voteCount": 1,
      }
    }
    movieTrailerKey="test"
    updateMovieTrailerKey={[MockFunction]}
  />
  <MovieCard
    genres={
      Array [
        Object {
          "id": 1,
          "name": "Action",
        },
        Object {
          "id": 2,
          "name": "SciFi",
        },
      ]
    }
    isGrid={true}
    movie={
      Object {
        "adult": false,
        "backdropPath": "/test.jpg",
        "genreIds": Array [
          1,
          2,
        ],
        "id": 1,
        "originalLanguage": "en",
        "originalTitle": "test",
        "overview": "test",
        "popularity": 1,
        "posterPath": "/test.jpg",
        "releaseDate": "1",
        "title": "test",
        "video": false,
        "voteAverage": 1,
        "voteCount": 1,
      }
    }
    movieTrailerKey="test"
    updateMovieTrailerKey={[MockFunction]}
  />
</section>
`;

exports[`render MovieList component render table layout correctly 1`] = `
<section
  className="movieListTable"
>
  <MovieCard
    genres={
      Array [
        Object {
          "id": 1,
          "name": "Action",
        },
        Object {
          "id": 2,
          "name": "SciFi",
        },
      ]
    }
    isGrid={false}
    movie={
      Object {
        "adult": false,
        "backdropPath": "/test.jpg",
        "genreIds": Array [
          1,
          2,
        ],
        "id": 1,
        "originalLanguage": "en",
        "originalTitle": "test",
        "overview": "test",
        "popularity": 1,
        "posterPath": "/test.jpg",
        "releaseDate": "1",
        "title": "test",
        "video": false,
        "voteAverage": 1,
        "voteCount": 1,
      }
    }
    movieTrailerKey="test"
    updateMovieTrailerKey={[MockFunction]}
  />
  <MovieCard
    genres={
      Array [
        Object {
          "id": 1,
          "name": "Action",
        },
        Object {
          "id": 2,
          "name": "SciFi",
        },
      ]
    }
    isGrid={false}
    movie={
      Object {
        "adult": false,
        "backdropPath": "/test.jpg",
        "genreIds": Array [
          1,
          2,
        ],
        "id": 1,
        "originalLanguage": "en",
        "originalTitle": "test",
        "overview": "test",
        "popularity": 1,
        "posterPath": "/test.jpg",
        "releaseDate": "1",
        "title": "test",
        "video": false,
        "voteAverage": 1,
        "voteCount": 1,
      }
    }
    movieTrailerKey="test"
    updateMovieTrailerKey={[MockFunction]}
  />
</section>
`;
